standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/JdoPropertyStore.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+import org.apache.commons.jexl3.JexlException;
+import org.apache.hadoop.hive.metastore.RawStore;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.model.MMetastoreDBProperties;
+import org.apache.hadoop.hive.metastore.properties.PropertyException;
+import org.apache.hadoop.hive.metastore.properties.PropertyMap;
+import org.apache.hadoop.hive.metastore.properties.PropertySchema;
+import org.apache.hadoop.hive.metastore.properties.PropertyStore;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Implementation of the property store delegating persistence to a (jdo) raw store.
+ */
+public class JdoPropertyStore extends PropertyStore {
+  /** The jdo objects store. */
+  private final RawStore objectStore;
+
+  /**
+   * Basic ctor.
+   * @param store the object store
+   */
+  public JdoPropertyStore(RawStore store) {
+    this.objectStore = store;
+  }
+
+  @Override
+  public PropertyMap fetchProperties(final String mapKey, Function<String, PropertySchema> getSchema) {
+    try {
+      return objectStore.getProperties(mapKey, getPropertyMapFunction(null, getSchema));
```

Comment:
we can move the logic of `objectStore.getProperties` to here

Reply:
The logic is using the persistence manager of the object store; trying to respect the encapsulation...